include/dsn/cpp/serialization_helper/thrift_helper.h

```diff
@@ -530,58 +530,14 @@ namespace dsn {
     inline uint32_t task_code::read(apache::thrift::protocol::TProtocol *iprot)
     {
         std::string task_code_string;
-        uint32_t xfer = 0;
-        apache::thrift::protocol::TBinaryProtocol* binary_proto = dynamic_cast<apache::thrift::protocol::TBinaryProtocol*>(iprot);
-        if (binary_proto != nullptr)
-        {
-            //the protocol is binary protocol
-            xfer += iprot->readString(task_code_string);
-        }
-        else
-        {
-            //the protocol is json protocol
-            std::string fname;
-            ::apache::thrift::protocol::TType ftype;
-            int16_t fid;
-
-            xfer += iprot->readStructBegin(fname);
-
-            using ::apache::thrift::protocol::TProtocolException;
-
-
-            while (true)
-            {
-                xfer += iprot->readFieldBegin(fname, ftype, fid);
-                if (ftype == ::apache::thrift::protocol::T_STOP) {
-                    break;
-                }
-                switch (fid)
-                {
-                case 1:
-                    if (ftype == ::apache::thrift::protocol::T_STRING) {
-                        xfer += iprot->readString(task_code_string);
-                    }
-                    else {
-                        xfer += iprot->skip(ftype);
-                    }
-                    break;
-                default:
-                    xfer += iprot->skip(ftype);
-                    break;
-                }
-                xfer += iprot->readFieldEnd();
-            }
-
-            xfer += iprot->readStructEnd();
-        }
+        uint32_t xfer = iprot->readString(task_code_string);
```

Comment:
I really don't like this because the serialization method is different for binary and json. Later I will try to modify your generated js code and check if this works. And another question which I asked many times, have you tried the TBinaryProtocol for javascript? If that doesn't work, I will stop to argue this.